File: from-0901-to-1000/0974-subarray-sums-divisible-by-k/solution.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        remainders = [0] * k
        remainders[0] = 1
        psum = 0
        result = 0
        for num in nums:
            psum += num
            rem = psum % k
            result += remainders[rem]
            remainders[rem] += 1

        return result

File: from-0901-to-1000/0974-subarray-sums-divisible-by-k/solution.go
func subarraysDivByK(nums []int, k int) int {
    mods := make(map[int]int)
    mods[0] = 1
    rem := 0
    result := 0
    for _, num := range nums {
        rem = (((rem + num) % k) + k) % k
        result += mods[rem]
        mods[rem] += 1
    }
    return result
}